documents/coredocument.go

```diff
@@ -75,6 +78,7 @@ func newCoreDocument() *CoreDocument {
 		DocumentIdentifier: id,
 		CurrentVersion:     id,
 		NextVersion:        utils.RandomSlice(idSize),
+		SignatureData:      new(coredocumentpb.SignatureData),
```

Comment:
we don't need this if we checking for nil in `AppendSignatures` methods, right ?